instructure/canvas-ios
# Fix annotation saving

refs: MBL-15939
affects: Student, Teacher
release note: Fixed annotations not getting saved correctly in some special cases.

test plan:
- See ticket for detailed description on the original issue.
- The upload mechanism was redesigned so the original repro steps don't apply here.
- Setup Charles to provide a slow networks. Proxy > Throttle settings (e.g.: 256kbps modem, stability 50%, unstable quality range 20-30%).
- Setup a Charles breakpoint on /2018-03-07/sessions/*.
- Play around with annotations using the slow network and occasionally break on a request and abort it.
- The goal is that when you see the 'All annotations saved' message and close then reopen the submission, all annotations should be loaded back.
